packages/block-library/src/block/edit.js

```diff
@@ -132,12 +135,56 @@ function getOverridesFromBlocks( blocks, defaultValues ) {
 	return Object.keys( overrides ).length > 0 ? overrides : undefined;
 }
 
+function setBlockEditMode( setEditMode, blocks ) {
+	blocks.forEach( ( block ) => {
+		const editMode = isPartiallySynced( block )
+			? 'contentOnly'
+			: 'disabled';
+		setEditMode( block.clientId, editMode );
+		setBlockEditMode( setEditMode, block.innerBlocks );
+	} );
+}
+
+function editSourcePattern(
+	setRenderingMode,
+	patternId,
+	createSuccessNotice,
+	defaultRenderingMode
+) {
+	const currentArgs = getQueryArgs( window.location.href );
+	const currentUrlWithoutArgs = removeQueryArgs(
+		window.location.href,
+		...Object.keys( currentArgs )
+	);
+	const newUrl = addQueryArgs( currentUrlWithoutArgs, {
+		...currentArgs,
+		patternId,
+	} );
```

Comment:
It didn't seem worth adding in `@wordpress/router` dependency and the router context provider just for this one small push to the query params. I thought adding the pattern id to url would make the browser back button work, but it doesn't so I am wondering if we should add the pattern id to the store somehow instead, eg. `setEditedPattern` to track this between modes instead of appending to url - what do you think?